File: app/src/main/java/com/adenilson/testeandroid/base/BaseFragment.java
package com.adenilson.testeandroid.base;

import android.support.v4.app.Fragment;

/**
 * @author: Adenilson N. da Silva Junior
 * On date: 09/06/2018
 */

public abstract class BaseFragment extends Fragment implements BaseView {

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }

    @Override
    public void showError(int messageResourceId) {

    }
}

File: app/src/main/java/com/vctapps/santanderchallenge/core/presentation/BaseFragment.kt
package com.vctapps.santanderchallenge.core.presentation

import android.app.Activity
import android.content.Context
import android.support.v4.app.Fragment
import dagger.android.support.AndroidSupportInjection

abstract class BaseFragment: Fragment(), BaseView {

    @SuppressWarnings("deprecation")
    override fun onAttach(activity: Activity?) {
        AndroidSupportInjection.inject(this)
        super.onAttach(activity)
    }

    override fun onAttach(context: Context?) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun showLoading() {

    }

    override fun hideLoading() {

    }

    override fun showError() {

    }

    override fun hideError() {

    }

}